modules/tensor_mechanics/src/materials/ComputeExtraStressVDWGas.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeExtraStressVDWGas.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeExtraStressVDWGas);
+
+template <>
+InputParameters
+validParams<ComputeExtraStressVDWGas>()
+{
+  InputParameters params = validParams<ComputeExtraStressBase>();
+  params.addClassDescription(
+      "Computes a hydrostatic stress corresponding to the pressure of a van der Waals gas that is "
+      "added as an extra_stress to the stress computed by the constitutive model");
+  params.addRequiredParam<MaterialPropertyName>(
+      "b", "Hard-sphere exclusion volume of van der Waals gas atoms in nm^3");
+  params.addRequiredParam<MaterialPropertyName>("Va", "Atomic volume of lattice atoms in nm^3");
+  params.addRequiredParam<MaterialPropertyName>("T", "Temperature in K");
+  params.addRequiredCoupledVar("cg", "Gas concentration (relative to lattice atoms)");
+  params.addParam<Real>("nondim_factor",
+                        1.0,
+                        "Optional factor to non-dimensionalize pressure (pressure is calculated in "
+                        "Pa, set this factor to characteristic energy density used for "
+                        "non-dimensionalization if desired)");
```

Comment:
Just a comment, but I would have opted to have kB be an input parameter with a sensible default (and the inits in the documentation referring to that default).